packages/@cdktf/provider-generator/lib/get/generator/models/attribute-model.ts

```diff
@@ -135,15 +135,18 @@ export class AttributeModel {
 
   public get mapType(): string {
     const type = this.type;
-    if (type.isStringMap) {
+    if (type.isStringMap || (type.isMap && type.innerType === "string")) {
```

Comment:
It doesn't actually match that, but I agree that it should.
I've been contemplating a bigger refactor to how the types are represented, but have been holding off. I'll won't put anything big in this PR, but I'll make a couple small updates to clarify.

Reply:
Looking forward to that bigger refactor <3 I hope we can get it to a state where changing a return type or so can happen in one instead of 5 places :D 

Thank you for addressing this while keeping the scope small 👍